### PR TITLE
ME-12 Buyer Onboarding - qualify for net terms

### PR DESCRIPTION
**Pull request by dhiraj-toptal**
Added buyer onboarding logic and Qualify for net terms functionality.
Iframe added for a buyer to register for net terms and net terms limit is displayed once buyer is successfully enabled for net terms.

### CODE REVIEW

**Comment by ynaxonBalance** [comment]
@dhiraj-toptal Please add description so I understand the context of this PR

**Comment by dhiraj-toptal** [comment]
@ynaxonBalance  Added description for PR, can you review?

File: Block/Createbuyer.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use Magento\Framework\Pricing\Helper\Data;
+use Magento\Framework\View\Element\Html\Link;
+use Balancepay\Balancepay\Helper\Data as BalancepayHelper;
+
+/**
+ * Class Createbuyer
+ *
+ * Balancepay\Balancepay\Block
+ */
+class Createbuyer extends Link
```
Comment (ynaxonBalance) [inline Block/Createbuyer.php]: rename to camel case `CreateBuyer` and consider renaming to `CreateBuyerLink` so it will be more clear

File: Controller/Buyer/Creditlimit.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+namespace Balancepay\Balancepay\Controller\Buyer;
+
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\Controller\Result\JsonFactory;
+
+class Creditlimit extends Action
```
Comment (ynaxonBalance) [inline Controller/Buyer/Creditlimit.php]: rename to camel case CreditLimit

File: CustomerData/CustomSection.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\CustomerData\SectionSourceInterface;
+use Magento\Customer\Model\Session;
+use Magento\Framework\Pricing\Helper\Data;
+
+/**
+ * Class CustomSection
+ * @package Balancepay\Balancepay\CustomerData
+ */
+class CustomSection implements SectionSourceInterface
```
Comment (ynaxonBalance) [inline CustomerData/CustomSection.php]: What is this section? can you come up with more descriptive class name?

File: CustomerData/CustomSection.php [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        $customerId = $this->balancepayHelper->getCustomerSessionId();
+        if (!empty($customerId)) {
+            $customer = $this->customerRepositoryInterface->getById($customerId);
+            $customerAttributeData = $customer->__toArray();
+            $buyerId = isset($customerAttributeData['custom_attributes']['buyer_id']) ?
+                $customerAttributeData['custom_attributes']['buyer_id']['value'] : '';
+            $this->customerSession->setBuyerId($buyerId);
+            return $buyerId;
+        }
+        return 0;
```
Comment (ynaxonBalance) [inline CustomerData/CustomSection.php]: return `null`

File: CustomerData/CustomSection.php [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    /**
+     * @param $price
+     * @return float|string
+     */
+    public function formattedAmount($price)
```
Comment (ynaxonBalance) [inline CustomerData/CustomSection.php]: is this method used outside this class? if not consider to make private / protected

File: CustomerData/CustomSection.php (hunk 4 above)
Comment (ynaxonBalance) [reply CustomerData/CustomSection.php]: is this method used outside this class? if not consider to make private / protected

File: Model/Request/Buyers.php [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+use Balancepay\Balancepay\Model\Config;
+use Balancepay\Balancepay\Lib\Http\Client\Curl;
+use Balancepay\Balancepay\Helper\Data as HelperData;
+use Magento\Customer\Model\Session;
+use Magento\Customer\Model\Address;
+
+/**
+ * Balancepay webhooks request model.
+ */
+class Buyers extends AbstractRequest
```
Comment (ynaxonBalance) [inline Model/Request/Buyers.php:19]: Maybe rename to `BuyersTopicWebhookRequest`?

File: Model/Response/Buyers.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<?php
+namespace Balancepay\Balancepay\Model\Response;
+
+use Balancepay\Balancepay\Model\AbstractResponse;
+
+/**
+ * Balancepay webhooks response model.
+ */
+class Buyers extends AbstractResponse
```
Comment (ynaxonBalance) [inline Model/Response/Buyers.php:9]: Is this related to the webhook? if so, consider also renaming to something like `BuyersTopicWebhookResponse`

File: view/frontend/web/js/section.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    'jquery',
+    'uiComponent',
+    'Magento_Customer/js/customer-data',
+    'mage/url'
+], function ($, Component, customerData, urlBuilder) {
+    'use strict';
+
+    return Component.extend({
+        initialize: function () {
+            console.log('initialize');
```
Comment (ynaxonBalance) [inline view/frontend/web/js/section.js]: remove console log

File: view/frontend/web/js/section.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+                                messages: [{
+                                    text: 'There was a problem generating a qualification link. Please contact the administrator.',
+                                    type: 'error'
+                                }]
+                            });
+                        },1000);
+                    }
+                },
+                error: function (xhr, status, errorThrown) {
+                    console.log('Error happens. Try again.');
```
Comment (ynaxonBalance) [inline view/frontend/web/js/section.js]: remove console log

Comment (ynaxonBalance) [reply view/frontend/web/js/section.js]: Don't we need to handle error here?

**Review by ynaxonBalance — APPROVED** [review]
Looking good :) left some comments

File: Block/Createbuyer.php (hunk 1 above)
Comment (dhiraj-toptal) [reply Block/Createbuyer.php]: Renamed

File: Controller/Buyer/Creditlimit.php (hunk 2 above)
Comment (dhiraj-toptal) [reply Controller/Buyer/Creditlimit.php]: Renamed

File: CustomerData/CustomSection.php (hunk 3 above)
Comment (dhiraj-toptal) [reply CustomerData/CustomSection.php]: Updated

File: CustomerData/CustomSection.php (hunk 4 above)
Comment (dhiraj-toptal) [reply CustomerData/CustomSection.php]: In ME-29 PR this class has been updated and functions are removed for refactoring

File: CustomerData/CustomSection.php (hunk 5 above)
Comment (dhiraj-toptal) [reply CustomerData/CustomSection.php]: In ME-29 PR this class has been updated and functions are removed for refactoring

File: Model/Request/Buyers.php (hunk 6 above)
Comment (dhiraj-toptal) [reply Model/Request/Buyers.php:19]: This is not related to only webhook request, we also call differenct buyer related APIs using this

File: Model/Response/Buyers.php (hunk 7 above)
Comment (dhiraj-toptal) [reply Model/Response/Buyers.php:9]: This is not related to only webhook request, we also call differenct buyer related APIs using this

File: view/frontend/web/js/section.js (hunk 8 above)
Comment (dhiraj-toptal) [reply view/frontend/web/js/section.js]: Removed

File: view/frontend/web/js/section.js (hunk 9 above)
Comment (dhiraj-toptal) [reply view/frontend/web/js/section.js]: Displayed error message

**Comment by dhiraj-toptal** [comment]
@ynaxonBalance 
I have updated PR for ME-29 and managed to fix above review comments in it as well. 
Can you review that?
